internal/updater/download.go

```diff
@@ -0,0 +1,119 @@
+package updater
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/gopasspw/gopass/pkg/ctxutil"
+	"github.com/gopasspw/gopass/pkg/debug"
+	"github.com/gopasspw/gopass/pkg/termio"
+	"github.com/pkg/errors"
+)
+
+func extract(archive, dest string) error {
+	debug.Log("Reading from %s", archive)
+	fh, err := os.Open(archive)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = fh.Close()
+	}()
+
+	dfh, err := os.OpenFile(dest, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0755)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to open file: %s", dest)
+	}
+	defer func() {
+		_ = dfh.Close()
+	}()
+
+	gzr, err := gzip.NewReader(fh)
+	if err != nil {
+		return err
+	}
+	tarReader := tar.NewReader(gzr)
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return errors.Wrapf(err, "Failed to read from tar file")
+		}
+		name := filepath.Base(header.Name)
+		if header.Typeflag == tar.TypeReg && name == "gopass" {
+			_, err := io.Copy(dfh, tarReader)
+			return errors.Wrapf(err, "Failed to read gopass from tar file")
+		}
+	}
+	return errors.Errorf("file not found in archive")
+}
+
+func tryDownload(ctx context.Context, dest, url string) error {
```

Comment:
Isn't the download part done by `ghrel` already?

Reply:
No.